pkg/core/native/management.go

```diff
@@ -236,10 +264,10 @@ func (m *Management) Update(d dao.DAO, hash util.Uint160, neff *nef.File, manif
 	}
 	// if manifest was provided, update the contract manifest
 	if manif != nil {
-		contract.Manifest = *manif
-		if !contract.Manifest.IsValid(contract.Hash) {
+		if !manif.IsValid(contract.Hash) {
 			return nil, errors.New("invalid manifest for this contract")
 		}
+		contract.Manifest = *manif
```

Comment:
I think you should mark the transaction as updated before doing this. Suppose it fails, you still gonna have updated manifest (or script) in the `contract`.

Reply:
fixed.

Reply:
> updated manifest **(or script)**